essentials-framework/Essentials Core/PepperDashEssentialsBase/Display/DisplayBase.cs

```diff
@@ -1,281 +1,287 @@
+
+            var commMonitor = displayDevice as ICommunicationMonitor;
+            if (commMonitor != null)
+            {
+                commMonitor.CommunicationMonitor.IsOnlineFeedback.LinkInputSig(trilist.BooleanInput[joinMap.IsOnline.JoinNumber]);
+            }
+
+            var inputNumberFeedback = new IntFeedback(() => inputNumber);
+
+            // Two way feedbacks
+            var twoWayDisplay = displayDevice as TwoWayDisplayBase;
+
+            if (twoWayDisplay != null)
+            {
+                trilist.SetBool(joinMap.IsTwoWayDisplay.JoinNumber, true);
+
+                twoWayDisplay.CurrentInputFeedback.OutputChange += (o, a) => Debug.Console(0, "CurrentInputFeedback_OutputChange {0}", a.StringValue);
+
+
+                inputNumberFeedback.LinkInputSig(trilist.UShortInput[joinMap.InputSelect.JoinNumber]);
+            }
+
+            // Power Off
+            trilist.SetSigTrueAction(joinMap.PowerOff.JoinNumber, () =>
+            {
+                inputNumber = 102;
+                inputNumberFeedback.FireUpdate();
+                displayDevice.PowerOff();
+            });
+
+            displayDevice.PowerIsOnFeedback.OutputChange += (o, a) =>
+            {
+                if (!a.BoolValue)
+                {
+                    inputNumber = 102;
+                    inputNumberFeedback.FireUpdate();
+
+                }
+                else
+                {
+                    inputNumber = 0;
+                    inputNumberFeedback.FireUpdate();
+                }
+            };
+
+            displayDevice.PowerIsOnFeedback.LinkComplementInputSig(trilist.BooleanInput[joinMap.PowerOff.JoinNumber]);
+
+            // PowerOn
+            trilist.SetSigTrueAction(joinMap.PowerOn.JoinNumber, () =>
+            {
+                inputNumber = 0;
+                inputNumberFeedback.FireUpdate();
+                displayDevice.PowerOn();
+            });
+
+
+            displayDevice.PowerIsOnFeedback.LinkInputSig(trilist.BooleanInput[joinMap.PowerOn.JoinNumber]);
+
+            for (int i = 0; i < displayDevice.InputPorts.Count; i++)
+            {
+                if (i < joinMap.InputNamesOffset.JoinSpan)
+                {
+                    inputKeys.Add(displayDevice.InputPorts[i].Key);
+                    var tempKey = inputKeys.ElementAt(i);
+                    trilist.SetSigTrueAction((ushort)(joinMap.InputSelectOffset.JoinNumber + i),
+                        () => displayDevice.ExecuteSwitch(displayDevice.InputPorts[tempKey].Selector));
+                    Debug.Console(2, displayDevice, "Setting Input Select Action on Digital Join {0} to Input: {1}", 
+                        joinMap.InputSelectOffset.JoinNumber + i, displayDevice.InputPorts[tempKey].Key.ToString());
+                    trilist.StringInput[(ushort)(joinMap.InputNamesOffset.JoinNumber + i)].StringValue = displayDevice.InputPorts[i].Key.ToString();
+                }
+            }

```

Comment:
This is looking good.  Can we add an else condition that prints a debug statement to warn users if they have a display with more inputs than the span?

Reply:
Will Do!

Reply:
Added a debug statement that says 
`"Device has {0} inputs.  The Join Map allows up to {1} inputs.  Discarding inputs {2} - {3} from bridge."` 
while displaying pertinent info, obviously

Reply:
I also want to note that I DID add MuteOn and MuteOff discrete controls to the joinMap and Bridge - they always get added when we build plugins anyway.  Shouldn't break anything.

Reply:
This is SUPER close.  Only change I would make is to set the level for the debug statement to 0 and use the Debug.Console() overload that logs the message to the error log at the warning level as well.  This is the kind of thing that ins't just for debugging but we actually want to let the user take note of when they can't figure out why their input switching isn't working.

Reply:
Done 

`Debug.Console(0, displayDevice, Debug.ErrorLogLevel.Warning, "Device has {0} inputs.  The Join Map allows up to {1} inputs.  Discarding inputs {2} - {3} from bridge.",
                        displayDevice.InputPorts.Count, joinMap.InputNamesOffset.JoinSpan, i + 1, displayDevice.InputPorts.Count);`